src/lib.rs

```diff
@@ -302,6 +302,22 @@
 //! If you would like to use a custom string parser other than `FromStr`, see
 //! the [same titled section](#custom-string-parsers) below.
 //!
+//! **Note:**
+//! _________________
+//! Pay attention that *only literal occurrence* of this types is special, for example
+//! `Option<T>` is special while `::std::option::Option<T>` is not.
+//!
+//! If you need to avoid special casing you can make a `type` alias and
+//! use it in place of the said type.
+//! _________________
+//!
+//! **Note:**
+//! _________________
+//! `bool` cannot be used as positional argument with an implicit parser. If you need
+//! a positional `bool`, for example to parse `true` or `false` add an explicit
+//! [`#[structopt(parse(...))]` annotation](#custom-string-parsers).
```

Comment:
If you need a positional `bool`, for example to parse `true` or `false`, you can add an explicit [`#[structopt(parse(try_from_str))]` annotation](#custom-string-parsers).